superset/assets/javascripts/explorev2/components/ExploreViewContainer.jsx

```diff
@@ -1,16 +1,55 @@
+/* eslint camelcase: 0 */
 import React from 'react';
+import { bindActionCreators } from 'redux';
+import * as actions from '../actions/exploreActions';
+import { connect } from 'react-redux';
 import ChartContainer from './ChartContainer';
 import ControlPanelsContainer from './ControlPanelsContainer';
 import QueryAndSaveBtns from '../../explore/components/QueryAndSaveBtns';
+const $ = require('jquery');
 
-export default class ExploreViewContainer extends React.Component {
+const propTypes = {
+  form_data: React.PropTypes.object.isRequired,
+  actions: React.PropTypes.object.isRequired,
+  slice_id: React.PropTypes.number.isRequired,
+  slice_name: React.PropTypes.string.isRequired,
+  datasource_id: React.PropTypes.number.isRequired,
+  datasource_type: React.PropTypes.string.isRequired,
+};
+
+
+class ExploreViewContainer extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       height: this.getHeight(),
     };
   }
 
+  onQuery() {
+    const data = {};
+    const form_data = this.props.form_data;
+    Object.keys(form_data).forEach((field) => {
+      // filter out null fields
```

Comment:
is there a case where a value might be set, but the user wants to set it back to null/no value?